Userland/Libraries/LibJS/Runtime/RegExpLegacyStaticProperties.cpp

```diff
@@ -0,0 +1,158 @@
+#include <LibJS/Runtime/RegExpConstructor.h>
+#include <LibJS/Runtime/VM.h>
+#include <LibJS/Runtime/Value.h>
+namespace JS {
+
+void RegExpLegacyStaticProperties::invalidate()
+{
+    m_input = {};
+    m_last_match = {};
+    m_last_paren = {};
+    m_left_context = {};
+    m_right_context = {};
+    m_$1 = {};
+    m_$2 = {};
+    m_$3 = {};
+    m_$4 = {};
+    m_$5 = {};
+    m_$6 = {};
+    m_$7 = {};
+    m_$8 = {};
+    m_$9 = {};
+}
+
+// GetLegacyRegExpStaticProperty( C, thisValue, internalSlotName ), https://github.com/tc39/proposal-regexp-legacy-features#getlegacyregexpstaticproperty-c-thisvalue-internalslotname-
+ThrowCompletionOr<Value> get_legacy_regexp_static_property(VM& vm, RegExpConstructor& constructor, Value this_value, Optional<String> (RegExpLegacyStaticProperties::*property_getter)() const)
+{
+    // 1. Assert C is an object that has an internal slot named internalSlotName.
+
+    // 2. If SameValue(C, thisValue) is false, throw a TypeError exception.
+    if (!same_value(&constructor, this_value))
+        return vm.throw_completion<TypeError>(ErrorType::GetLegacyRegExpStaticPropertyThisValueMismatch);
+
+    // 3. Let val be the value of the internal slot of C named internalSlotName.
+    auto val = (constructor.legacy_static_properties().*property_getter)();
+
+    // 4. If val is empty, throw a TypeError exception.
+    if (!val.has_value())
+        return vm.throw_completion<TypeError>(ErrorType::GetLegacyRegExpStaticPropertyValueEmpty);
+
+    // 5. Return val.
+    return js_string(vm, val.value());
+}
+
+// SetLegacyRegExpStaticProperty( C, thisValue, internalSlotName, val ), https://github.com/tc39/proposal-regexp-legacy-features#setlegacyregexpstaticproperty-c-thisvalue-internalslotname-val-
+ThrowCompletionOr<void> set_legacy_regexp_static_property(VM& vm, RegExpConstructor& constructor, Value this_value, void (RegExpLegacyStaticProperties::*property_setter)(String const&), Value value)
+{
+    // 1. Assert C is an object that has an internal slot named internalSlotName.
+
+    // 2. If SameValue(C, thisValue) is false, throw a TypeError exception.
+    if (!same_value(&constructor, this_value))
+        return vm.throw_completion<TypeError>(ErrorType::SetLegacyRegExpStaticPropertyThisValueMismatch);
+
+    // 3. Let strVal be ? ToString(val).
+    auto str_value = TRY(value.to_string(vm));
+
+    // 4. Set the value of the internal slot of C named internalSlotName to strVal.
+    (constructor.legacy_static_properties().*property_setter)(move(str_value));
+
+    return {};
+}
+
+// UpdateLegacyRegExpStaticProperties ( C, S, startIndex, endIndex, capturedValues ), https://github.com/tc39/proposal-regexp-legacy-features#updatelegacyregexpstaticproperties--c-s-startindex-endindex-capturedvalues-
+void update_legacy_regexp_static_properties([[maybe_unused]] VM& vm, RegExpConstructor& constructor, Utf16String const& string, size_t start_index, size_t end_index, Vector<Utf16View> const& captured_values)
```

Comment:
Don't pass a VM if you don't need it

```suggestion
void update_legacy_regexp_static_properties(RegExpConstructor& constructor, Utf16String const& string, size_t start_index, size_t end_index, Vector<Utf16View> const& captured_values)
```